tests/test_databases.py

```diff
@@ -503,8 +515,13 @@ def run_sync(*args, **kwargs):
         async with database.transaction(force_rollback=True):
             # execute()
             query = session.insert()
-            values = {"data": {"text": "hello", "boolean": True, "int": 1}}
-            await database.execute(query, values)
+            data = {"text": "hello", "boolean": True, "int": 1}
+            values = {"data": data}
+
+            if database.url.scheme == "postgresql+aiopg":
+                await database.execute(query, {"data": json.dumps(data)})
```

Comment:
Can we handle this automatically for our users?
Would that be a sensible thing to do here?

Reply:
Not sure it's possible in all cases, as I would need to know that the column of a type json, so it'll require either a lookup to the sqlalchemy's metadata or to the DB schema. But I'll have a look at how easy it can be.

Reply:
> it'll require either a lookup to the sqlalchemy's metadata or to the DB schema

If it means a lookup then let's not do it. I'm just not clear why `JSONField` isn't already handling the serialization for us in this case. Might need some more looking into?

Reply:
yeah, I'm dealing with it now, there're several facets to approach it.
it can work canonically, but the field has to be `sqlalchemy.dialects.postgresql.JSON`, but `aiopg` returns back (over `select`) the decoded value and it breaks the `RowProxy` as it expects string.
so, i'm looking for compromise/settings to get the balance

Reply:
ok, it was the matter of the dialect class, I reused the one that's in use for the integration with `sa` part of `aoipg` it got back to normal.